src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/Json/ObjectReaders.cs

```diff
@@ -346,7 +347,9 @@ public static RazorProjectInfo ReadProjectInfoFromProperties(JsonDataReader read
         var rootNamespace = reader.ReadString(nameof(RazorProjectInfo.RootNamespace));
         var documents = reader.ReadImmutableArray(nameof(RazorProjectInfo.Documents), static r => r.ReadNonNullObject(ReadDocumentSnapshotHandleFromProperties));
 
-        return new RazorProjectInfo(serializedFilePath, filePath, configuration, rootNamespace, projectWorkspaceState, documents);
+        var displayName = Path.GetFileNameWithoutExtension(filePath);
```

Comment:
@DustinCampbell it seemed like this method is only used in benchmarks and tests of the serializer, so I didn't bother making this actually work. Should we just remove the json serialization for RazorProjectInfo? Or are the json files still used for tag helper info in other tests?

Reply:
Maybe? There's still test code that reads JSON test data, which is still convenient.